frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/RowActionsDropdown.tsx

```diff
@@ -1,26 +1,73 @@
+import { Dispatch, SetStateAction, useState } from 'react'
 import {
+  BiChevronLeft,
+  BiChevronRight,
   BiDuplicate,
+  BiFolder,
   BiShareAlt,
   BiShow,
   BiTrash,
   BiUserPlus,
 } from 'react-icons/bi'
 import { Link as ReactLink } from 'react-router-dom'
-import { ButtonGroup, MenuButton, MenuDivider } from '@chakra-ui/react'
+import { ButtonGroup, Icon, MenuButton, MenuDivider } from '@chakra-ui/react'
+
+import { AdminDashboardFormMetaDto } from '~shared/types'
+import { Workspace } from '~shared/types/workspace'
 
 import { BxsChevronDown } from '~assets/icons/BxsChevronDown'
 import { BxsChevronUp } from '~assets/icons/BxsChevronUp'
 import Button from '~components/Button'
 import IconButton from '~components/IconButton'
 import Menu from '~components/Menu'
 
+import { useWorkspaceContext } from '~features/workspace/WorkspaceContext'
+
 import { RowActionsProps } from './RowActions'
 import { useRowAction } from './useRowAction'
 
+const MoveWorkspaceDropdown = ({
+  setIsMoveWorkspace,
+  formMeta,
+}: {
+  setIsMoveWorkspace: Dispatch<SetStateAction<boolean>>
+  formMeta: AdminDashboardFormMetaDto
+}) => {
+  const { handleMoveForm } = useRowAction(formMeta)
+  const { workspaces } = useWorkspaceContext()
+
+  const handleWorkspaceClick = (destWorkspace: Workspace) =>
+    handleMoveForm(destWorkspace._id.toString(), destWorkspace.title)
```

Comment:
Should use a `useCallback` here (might be more efficient).

Reply:
Thanks @kenjin-work! [ad5b50e](https://github.com/opengovsg/FormSG/pull/6560/commits/ad5b50e7937face50761c8b0d1f84f68d3ac0c4c)